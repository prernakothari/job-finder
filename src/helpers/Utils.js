export function TimeAgo(created_at, currentTime = undefined) {
    created_at = created_at.split(" ")
    let month = {
        "Jan": 0,
        "Feb": 1,
        "Mar": 2,
        "Apr": 3,
        "May": 4,
        "Jun": 5,
        "Jul": 6,
        "Aug": 7,
        "Sep": 8,
        "Oct": 9,
        "Nov": 10,
        "Dec": 11
    }
    let minutes = 1000 * 60;
    let hours = minutes * 60;
    let days = hours * 24;
    let months = days * 30;
    let years = days * 365;
    let createdDayTime = created_at[3].split(":").map(it => Number(it))
    if (currentTime === undefined)
        currentTime = new Date().getTime()

    let createdDate = Date.UTC(Number(created_at[5]), month[created_at[1]], Number(created_at[2]), createdDayTime[0], createdDayTime[1], createdDayTime[2]);
    let deltaT = currentTime - createdDate

    if (deltaT < 1000 * 60)
        return "Just now"
    else if (deltaT > minutes && deltaT < hours)
        return `${Math.floor(deltaT / minutes)} min ago`
    else if (deltaT > hours && deltaT < days)
        return `${Math.floor(deltaT / hours)}h ago`
    else if (deltaT > days && deltaT < months)
        return `${Math.floor(deltaT / days)}d ago`
    else if (deltaT > months && deltaT < years)
        return `${Math.floor(deltaT / months)} months ago`
    else
        return `${Math.floor(deltaT / years)}yr ago`
}


