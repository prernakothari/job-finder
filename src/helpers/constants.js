export const colors = {
    bgLight: "#F5F6F8",
    bgDark: "#131822",
    mainLight: "#FFFFFF",
    mainDark: "#19212D",
    purple: "#5865E0",
    darkPurple: "#3D469C",
    lightPurple: "#CECFF6"
}