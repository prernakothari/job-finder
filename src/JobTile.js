import React, { useEffect } from "react"
import { makeStyles, withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import Avatar from '@material-ui/core/Avatar';
import "./JobTile.css";
import { TimeAgo } from "./Utils"
import { useHistory } from "react-router-dom";
import { colors } from "./constants"
// is location is a number (US zip code), we could use zip static for obtaining City, State in US


export default function JobTile({ themeType, title, type, created_at, company_logo, company, location, url }) {
    const useStyles = makeStyles((theme) => ({
        root: {
            textAlign: "left",
            minWidth: 275,
            position: "relative",
            top: - theme.spacing(3.5),
            '&:hover': {
                "cursor": "pointer"
            },
            backgroundColor: themeType === "light" ? colors.mainLight : colors.mainDark,
            color: themeType === "light" ? "black" : "white",
            [theme.breakpoints.up('sm')]: {
                height: 225,
            },
        },
        bullet: {
            display: 'inline-block',
            margin: '0 2px',
            transform: 'scale(0.8)',
        },
        title: {
            marginTop: theme.spacing(3.5),
            fontSize: 14,
            color: themeType === "light" ? colors.mainDark : colors.mainLight,
        },
        pos: {
            color: themeType === "light" ? "black" : "white",
            [theme.breakpoints.down('xs')]: {
                marginBottom: theme.spacing(3),
            },
        },
        location: {
            color: colors.purple,
            position: "absolute",
            bottom: theme.spacing(2)
        },
        logo: {
            backgroundColor: "white",
            marginLeft: 16,
            height: theme.spacing(7),
            width: theme.spacing(7),
            zIndex: 2,
            boxShadow: `0px 2px 1px -1px rgb(0 0 0 / 20%), 0px 1px 1px 0px rgb(0 0 0 / 14%), 0px 1px 3px 0px rgb(0 0 0 / 12%); `
        },

    }));


    let timeAgo = TimeAgo(created_at)
    const classes = useStyles();
    const bull = <span className={classes.bullet}>â€¢</span>;
    const history = useHistory();
    const handleClick = () => {
        history.push("/positions/" + url.split("/").pop());
    }

    return (
        <div className={classes.main}>
            <Avatar variant="rounded" src={company_logo} sx={{ ".MuiAvatar-img": "contain" }} className={classes.logo} />
            <Card className={classes.root} onClick={handleClick}>
                <CardContent >
                    <Typography className={classes.title} gutterBottom>
                        {timeAgo}   {bull}   {type}
                    </Typography>
                    <Typography variant="h6" component="h6">
                        {title}
                    </Typography>
                    <Typography className={classes.pos}>
                        {company}
                    </Typography>
                    <Typography className={classes.location}>
                        {location}
                    </Typography>
                </CardContent>
            </Card>
        </div>
    );
}
