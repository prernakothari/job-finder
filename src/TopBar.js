import React from 'react';
import { Link } from "react-router-dom";
import { makeStyles, withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import FormGroup from '@material-ui/core/FormGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Switch from '@material-ui/core/Switch';
import { purple } from '@material-ui/core/colors';
import Grid from '@material-ui/core/Grid';
import WbSunnyIcon from '@material-ui/icons/WbSunny';
import Brightness3Icon from '@material-ui/icons/Brightness3';
import { colors } from "./constants";


export default function TopBar({ themeType, handleThemeChange }) {
    const useStyles = makeStyles((theme) => ({
        root: {
            flexGrow: 1,
            height: theme.spacing(14),
            backgroundColor: themeType === "light" ? colors.bgLight : colors.bgDark,
            "box-shadow": "none",
        },
        homeNav: {
            color: "#FFFFFF",
            "text-decoration": "none"
        },
        toolbar: {
            height: theme.spacing(14),
            backgroundColor: colors.purple,
            "border-bottom-left-radius": "55px",
            [theme.breakpoints.down('xs')]: {
                "border-bottom-left-radius": "0px"
            }
        },
        title: {
            flexGrow: 1,
        },
    }));

    const ThemeSwitch = withStyles({
        switchBase: {
            color: "gray",
            '&$checked': {
                color: "white",
            },
            '&$checked + $track': {
                backgroundColor: colors.darkPurple,
            },
        },
        checked: {},
        track: {},
    })(Switch);

    const classes = useStyles();
    const [state, setState] = React.useState({ checkedA: false });

    const handleChange = (event) => {
        if (event.target.checked)
            handleThemeChange("dark")
        else
            handleThemeChange("light")

        setState({ ...state, [event.target.name]: event.target.checked });
    };

    return (
        <div className={classes.root}>
            <AppBar position="static" className={classes.root}>
                <Toolbar className={classes.toolbar}>
                    <Typography variant="h6" className={classes.title}>
                        <Link to={"/"} className={classes.homeNav} >devjobs</Link>
                    </Typography>
                    <Typography component="div">
                        <Grid component="label" container alignItems="center" spacing={1}>
                            <Grid item>
                                <WbSunnyIcon fontSize="small" />
                            </Grid>
                            <Grid item>
                                <ThemeSwitch size="small" checked={state.checkedA} onChange={handleChange} name="checkedA" />
                            </Grid>
                            <Grid item>
                                <Brightness3Icon fontSize="small" />
                            </Grid>
                        </Grid>
                    </Typography>
                </Toolbar>
            </AppBar>
        </div>
    );
}
