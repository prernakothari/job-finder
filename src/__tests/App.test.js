import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from "react-router-dom";
import TopBar from '../components/TopBar';
import JobTile from '../components/JobTile';
import { TimeAgo } from '../helpers/Utils';
import testData from './testData';

test('renders learn react link', () => {
  render(<Router>
    <TopBar />
  </Router>)
  const linkElement = screen.getByText("devjobs");
  expect(linkElement).toBeDefined()
});

test('test Time Ago method: T1', () => {
  let timeAgo = TimeAgo("Fri Apr 09 00:06:43 UTC 2021", 1617931686366)
  expect(timeAgo).toBe("1h ago")
})

test('test Time Ago method: T2', () => {
  let timeAgo = TimeAgo("Thu Apr 01 19:07:46 UTC 2021", 1617931686366)
  expect(timeAgo).toBe("7d ago")
})

it('JobTile renders correctly', () => {
  let item = testData()[0]
  const tree = renderer
    .create(<JobTile themeType={'light'} {...item} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('TopBar renders correctly', () => {
  let item = testData()[0]
  const tree = renderer
    .create(<Router> <TopBar /> </Router>).toJSON();
  expect(tree).toMatchSnapshot();
});