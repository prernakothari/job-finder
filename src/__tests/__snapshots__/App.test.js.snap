// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
Array [
  " ",
  <div
    style={
      Object {
        "backgroundColor": "#F5F6F8",
      }
    }
  >
    <div
      className="makeStyles-root-51"
    >
      <header
        className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-root-51 MuiPaper-elevation4"
      >
        <div
          className="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-53 MuiToolbar-gutters"
        >
          <h6
            className="MuiTypography-root makeStyles-title-54 MuiTypography-h6"
          >
            <a
              className="makeStyles-homeNav-52"
              href="/"
              onClick={[Function]}
            >
              devjobs
            </a>
          </h6>
          <div
            className="MuiTypography-root MuiTypography-body1"
          >
            <label
              className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-center"
            >
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6.76 4.84l-1.8-1.79-1.41 1.41 1.79 1.79 1.42-1.41zM4 10.5H1v2h3v-2zm9-9.95h-2V3.5h2V.55zm7.45 3.91l-1.41-1.41-1.79 1.79 1.41 1.41 1.79-1.79zm-3.21 13.7l1.79 1.8 1.41-1.41-1.8-1.79-1.4 1.4zM20 10.5v2h3v-2h-3zm-8-5c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6-2.69-6-6-6zm-1 16.95h2V19.5h-2v2.95zm-7.45-3.91l1.41 1.41 1.79-1.8-1.41-1.41-1.79 1.8z"
                  />
                </svg>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <span
                  className="MuiSwitch-root MuiSwitch-sizeSmall"
                >
                  <span
                    aria-disabled={false}
                    className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-25 MuiSwitch-switchBase WithStyles(ForwardRef(Switch))-switchBase-55 MuiSwitch-colorSecondary"
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={null}
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <input
                        checked={false}
                        className="PrivateSwitchBase-input-28 MuiSwitch-input"
                        name="checkedA"
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <span
                        className="MuiSwitch-thumb"
                      />
                    </span>
                    <span
                      className="MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    className="MuiSwitch-track WithStyles(ForwardRef(Switch))-track-57"
                  />
                </span>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9 2c-1.05 0-2.05.16-3 .46 4.06 1.27 7 5.06 7 9.54 0 4.48-2.94 8.27-7 9.54.95.3 1.95.46 3 .46 5.52 0 10-4.48 10-10S14.52 2 9 2z"
                  />
                </svg>
              </div>
            </label>
          </div>
        </div>
      </header>
    </div>
    <div
      className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 40,
          "left": "50%",
          "position": "absolute",
          "top": "50%",
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
    <div>
      <div
        className="makeStyles-root-86"
      >
        <div
          className="MuiContainer-root MuiContainer-fixed MuiContainer-maxWidthLg"
        >
          <div
            className="MuiPaper-root MuiCard-root makeStyles-desktopSearchBar-87 makeStyles-searchBar-89 MuiPaper-elevation1 MuiPaper-rounded"
            position="static"
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
            >
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-4 MuiGrid-grid-lg-4"
              >
                <div
                  className="makeStyles-search-90"
                >
                  <div
                    className="makeStyles-searchIcon-91"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                      />
                    </svg>
                  </div>
                  <div
                    className="MuiInputBase-root makeStyles-inputRoot-92 MuiInputBase-fullWidth"
                    onClick={[Function]}
                  >
                    <input
                      aria-label="search"
                      className="MuiInputBase-input makeStyles-inputInput-93"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder="Filter by title, companies, expertise..."
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <hr
                className="MuiDivider-root MuiDivider-flexItem MuiDivider-vertical"
              />
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-4 MuiGrid-grid-lg-4"
              >
                <div
                  className="makeStyles-search-90"
                >
                  <div
                    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
                  >
                    <div
                      className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-1 MuiGrid-grid-md-2"
                    >
                      <button
                        active={true}
                        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
                        disabled={false}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onDragLeave={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        tabIndex={0}
                        type="button"
                      >
                        <span
                          className="MuiButton-label"
                        >
                          <svg
                            aria-hidden={true}
                            className="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
                            />
                          </svg>
                        </span>
                        <span
                          className="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <div
                      className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-9 MuiGrid-grid-md-8"
                    >
                      <div
                        className="MuiInputBase-root makeStyles-inputRoot-92"
                        onClick={[Function]}
                      >
                        <input
                          aria-label="search"
                          className="MuiInputBase-input makeStyles-inputInput-93 makeStyles-inputLocation-94"
                          onAnimationStart={[Function]}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          placeholder="Filter by Location..."
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <hr
                className="MuiDivider-root MuiDivider-flexItem MuiDivider-vertical"
              />
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-4 MuiGrid-grid-lg-4"
                style={
                  Object {
                    "maxWidth": "32%",
                  }
                }
              >
                <div
                  className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
                >
                  <div
                    className="MuiGrid-root MuiGrid-item"
                  >
                    <label
                      className="MuiFormControlLabel-root makeStyles-checkboxForm-95"
                    >
                      <span
                        aria-disabled={false}
                        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-25 MuiCheckbox-root Component-root-98 MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                        onBlur={[Function]}
                        onDragLeave={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        tabIndex={null}
                      >
                        <span
                          className="MuiIconButton-label"
                        >
                          <input
                            checked={false}
                            className="PrivateSwitchBase-input-28"
                            data-indeterminate={false}
                            name="checkedA"
                            onChange={[Function]}
                            type="checkbox"
                          />
                          <svg
                            aria-hidden={true}
                            className="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                            />
                          </svg>
                        </span>
                      </span>
                      <span
                        className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                      >
                        Full Time Only
                      </span>
                    </label>
                  </div>
                  <div
                    className="MuiGrid-root makeStyles-searchButton-97 MuiGrid-item"
                  >
                    <button
                      className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-50 MuiButton-contained MuiButton-containedPrimary"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiButton-label"
                      >
                        Search
                      </span>
                      <span
                        className="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="MuiPaper-root MuiCard-root makeStyles-mobileSearchBar-88 makeStyles-searchBar-89 MuiPaper-elevation1 MuiPaper-rounded"
            position="static"
          >
            <div
              className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
            >
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7"
              >
                <div
                  className="makeStyles-search-90"
                >
                  <div
                    className="MuiInputBase-root makeStyles-inputRoot-92 MuiInputBase-fullWidth"
                    onClick={[Function]}
                  >
                    <input
                      aria-label="search"
                      className="MuiInputBase-input makeStyles-inputInput-93"
                      onAnimationStart={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      placeholder="Filter by keyword..."
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
              >
                <div
                  className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-flex-end"
                >
                  <div
                    className="MuiGrid-root MuiGrid-item"
                  >
                    <button
                      className="MuiButtonBase-root MuiButton-root MuiButton-text"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiButton-label"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M3 18h6v-2H3v2zM3 6v2h18V6H3zm0 7h12v-2H3v2z"
                          />
                        </svg>
                      </span>
                      <span
                        className="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <div
                    className="MuiGrid-root MuiGrid-item"
                  >
                    <button
                      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <span
                        className="MuiButton-label"
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                          />
                        </svg>
                      </span>
                      <span
                        className="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="infinite-scroll-component__outerdiv"
            style={Object {}}
          >
            <div
              className="infinite-scroll-component "
              style={
                Object {
                  "WebkitOverflowScrolling": "touch",
                  "height": "auto",
                  "overflow": "auto",
                }
              }
            >
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  " ",
]
`;

exports[`JobTile renders correctly 1`] = `
<div>
  <div
    className="MuiAvatar-root MuiAvatar-rounded makeStyles-logo-17"
  >
    <img
      className="MuiAvatar-img"
      src="https://jobs.github.com/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBcmVmIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--dd8d40ec5297b1e8077942db328cc98330bb227c/Rijksoverheid.jpg"
    />
  </div>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-12 MuiPaper-elevation1 MuiPaper-rounded"
    onClick={[Function]}
  >
    <div
      className="MuiCardContent-root"
    >
      <p
        className="MuiTypography-root makeStyles-title-14 MuiTypography-body1 MuiTypography-gutterBottom"
      >
        2d ago
           
        <span
          className="makeStyles-bullet-13"
        >
          •
        </span>
           
        Full Time
      </p>
      <h6
        className="MuiTypography-root MuiTypography-h6"
      >
        Digital forensic researcher mobile device hacking
      </h6>
      <p
        className="MuiTypography-root makeStyles-pos-15 MuiTypography-body1"
      >
        Rijksoverheid
      </p>
      <p
        className="MuiTypography-root makeStyles-location-16 MuiTypography-body1"
      >
        Den Haag
      </p>
    </div>
  </div>
</div>
`;

exports[`TopBar renders correctly 1`] = `
Array [
  " ",
  <div
    className="makeStyles-root-18"
  >
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-root-18 MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-20 MuiToolbar-gutters"
      >
        <h6
          className="MuiTypography-root makeStyles-title-21 MuiTypography-h6"
        >
          <a
            className="makeStyles-homeNav-19"
            href="/"
            onClick={[Function]}
          >
            devjobs
          </a>
        </h6>
        <div
          className="MuiTypography-root MuiTypography-body1"
        >
          <label
            className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-center"
          >
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6.76 4.84l-1.8-1.79-1.41 1.41 1.79 1.79 1.42-1.41zM4 10.5H1v2h3v-2zm9-9.95h-2V3.5h2V.55zm7.45 3.91l-1.41-1.41-1.79 1.79 1.41 1.41 1.79-1.79zm-3.21 13.7l1.79 1.8 1.41-1.41-1.8-1.79-1.4 1.4zM20 10.5v2h3v-2h-3zm-8-5c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6-2.69-6-6-6zm-1 16.95h2V19.5h-2v2.95zm-7.45-3.91l1.41 1.41 1.79-1.8-1.41-1.41-1.79 1.8z"
                />
              </svg>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <span
                className="MuiSwitch-root MuiSwitch-sizeSmall"
              >
                <span
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-25 MuiSwitch-switchBase WithStyles(ForwardRef(Switch))-switchBase-22 MuiSwitch-colorSecondary"
                  onBlur={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={null}
                >
                  <span
                    className="MuiIconButton-label"
                  >
                    <input
                      checked={false}
                      className="PrivateSwitchBase-input-28 MuiSwitch-input"
                      name="checkedA"
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <span
                      className="MuiSwitch-thumb"
                    />
                  </span>
                </span>
                <span
                  className="MuiSwitch-track WithStyles(ForwardRef(Switch))-track-24"
                />
              </span>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 2c-1.05 0-2.05.16-3 .46 4.06 1.27 7 5.06 7 9.54 0 4.48-2.94 8.27-7 9.54.95.3 1.95.46 3 .46 5.52 0 10-4.48 10-10S14.52 2 9 2z"
                />
              </svg>
            </div>
          </label>
        </div>
      </div>
    </header>
  </div>,
  " ",
]
`;
